src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -863,7 +869,18 @@ internal bool MethodGroupIsCompatibleWithDelegate(BoundExpression receiverOpt, b
             }
 
             var sourceConstantValue = source.ConstantValue;
-            if (sourceConstantValue == null || sourceConstantValue.IsBad)
+            if (sourceConstantValue == null)
+            {
+                if (conversion.Kind == ConversionKind.DefaultOrNullLiteral && source.Kind == BoundKind.DefaultLiteral)
+                {
+                    return destination.GetDefaultValue();
```

Comment:
>GetDefaultValue [](start = 39, length = 15)

Is this method new to this branch? Could you provide a link to its implementation?